[LingZiyann] iP

LingZiyann opened the pull request:
# Duke the duke
With duke, you can finally duke with the duke
-~~text-based~~Voice based
-Free of charge
-0 latency

> "Duke is the best product i've ever used in my whole life!" - trust me bro

All you need to do is:

1. [Download it from here](https://example.com).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you.

And it is **FREE**
## Features
- [ ] Manage task
- [ ] Manage chickens
- [ ] Manage ricer
- [ ]  😉 😉 😉

## Setting up
`print("hello world")`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LingZiyann committed:
Added Level 0 and Level 1

LingZiyann committed:
Add level 2

LingZiyann committed:
Add level 3

LingZiyann committed:
Added Level 4

LingZiyann committed:
Add level 4

LingZiyann committed:
Merge branch 'master' of https://github.com/LingZiyann/ip

LingZiyann committed:
Add A-TextUiTesting

LingZiyann committed:
Add level-5

LingZiyann committed:
Add Level-6

LingZiyann committed:
Add A-Enums

LingZiyann committed:
Add A-Enums again

LingZiyann committed:
Add Level-7

LingZiyann committed:
Add Level-8

LingZiyann committed:
Add A-More OOP

LingZiyann committed:
Merge branch 'add-gradle-support'

LingZiyann committed:
Add A-Gradle

LingZiyann committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/data

LingZiyann committed:
Add A-JUnit

LingZiyann committed:
Merge branch 'increments'

# Conflicts:
#	src/main/java/data

LingZiyann committed:
A-JavaDoc

LingZiyann committed:
Add A-CodingStandard

Change class variable names to camelCase

Changes were made mainly to classes inside Bot package

LingZiyann committed:
Add Level-9

Finding a specific task feature was added

ListManager listItem method modified to allow this,
 and new switch statement added to Parser to handle find commands

LingZiyann committed:
Merge tag 'A-JavaDoc'

LingZiyann committed:
Merge branch 'branch-Level-9'

LingZiyann committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Bot/ListManager.java

TY1Fan commented on src/main/java/Bot/Parser.java:
```diff
@@ -0,0 +1,159 @@
+        String firstWord = parts[0].toLowerCase();
+
+        switch (firstWord) {
+            case "bye":
```
I think case clause should not be indented.

TY1Fan commented on src/main/java/TaskType/TaskBuilder.java:
```diff
@@ -0,0 +1,82 @@
+
+
+
+    public TaskBuilder by(String by) {
```
I think it would be better to not give methods and fields the same name?

TY1Fan commented on src/main/java/Bot/DateConverter.java:
```diff
@@ -0,0 +1,115 @@
+     * @param input The date string to be converted.
+     * @return The formatted date string or an error message if the format is invalid.
+     */
+    public static String convertDate(String input) {
```
would it be better to name it as convertDateToString so that it becomes more specific?

TY1Fan commented on src/main/java/Bot/ListManager.java:
```diff
@@ -0,0 +1,71 @@
+     * Method that takes in a string and returns all tasks that includes that string. Can be used
+     * to list all items by just passing in a "" argument
+     */
+    public String listItems(String item) {
```
Nice use of verbs and plural for the naming of this method.

TY1Fan commented on src/main/java/Bot/DateConverter.java:
```diff
@@ -0,0 +1,115 @@
+     * @param input The date string to be checked.
+     * @return true if the input matches the date format, false otherwise.
+     */
+    private static boolean isDateFormat(String input) {
```
I like how you name your boolean fields and method. Very easy to tell that it is boolean.

TY1Fan reviewed (commented):
I feel that the naming conventions are mostly adhered to. :+1:
Your code looks very clean and easy to understand.
But some nitty gritty details to note on styling like your case clause indentations.

LingZiyann committed:
Add Level-10

LingZiyann committed:
Add assertions to project

There are several places where assertions can be used to ensure that devs do not introduce new bugs to the code

Currently assertions used in Parser to ensure that if a command method was called, it was actually typed by the user's input
Also added to FileManager to ensure that a file path should always be present when calling file methods

LingZiyann committed:
Merge pull request #1 from LingZiyann/branch-A-Assertions

Add assertions to project

LingZiyann committed:
Improve code readability and maintainability

The current code contains redundant conditional checks and lacks a consistent approach to resource management. Methods have duplicated logic, which reduces readability and increases the risk of errors.

To enhance code quality:

Simplify conditional checks using boolean variables, improving readability and reducing redundancy.
Implement proper resource management using try-with-resources to automatically close Scanner and FileWriter, preventing memory leaks.
Improve error handling by providing more descriptive messages, making debugging easier.
Refactor assertions and comments for clarity, making the code self-documenting and easier to understand.

LingZiyann committed:
Merge pull request #2 from LingZiyann/branch-A-CodeQuality

Improve code readability and maintainability

LingZiyann committed:
Currently, the system allows the addition of duplicate tasks, which can lead to redundancy when managing tasks.

To prevent this, add a new `isEqual` method in the `Task` class to check for equality between tasks.
 This method is called whenever a new task is added to determine if it is a duplicate.

By handling duplicates at the point of addition, we ensure that the task list remains clean and manageable.
This approach is efficient as it checks for duplicates only when necessary.

LingZiyann committed:
Change 'Duke' name to Chicken

LingZiyann committed:
Update JavaFX integration

LingZiyann committed:
Add A-UserGuide

LingZiyann committed:
Add javadocs

LingZiyann committed:
Update package name to fit convention

LingZiyann committed:
Update Save feature and TaskTest

LingZiyann committed:
Update ReadMe

LingZiyann committed:
Fix issues with building of Jar file

LingZiyann committed:
Update 'Duke' to 'Chicken'

LingZiyann committed:
Update README
